File: src/AIClient/shell/src/command/positional_argument_builder.rs
use crate::command::PositionalArgument;

#[derive(Default)]
pub struct PositionalArgumentBuilder {
    argument: PositionalArgument,
}

impl PositionalArgumentBuilder {
    pub fn new() -> Self {
        PositionalArgumentBuilder::default()
    }

    pub fn name(mut self, name: &str) -> Self {
        self.argument.name = name.to_string();

        self
    }

    pub fn description(mut self, description: &str) -> Self {
        self.argument.description = description.to_string();

        self
    }

    pub fn default_value(mut self, default: &str) -> Self {
        self.argument.default = default.to_string();

        self
    }

    pub fn required(mut self) -> Self {
        self.argument.required = true;

        self
    }

    pub fn build(self) -> PositionalArgument {
        self.argument
    }
}

File: libs/Shell/PositionalArgumentBuilder.cpp
/*
** EPITECH PROJECT, 2026
** Shell
** File description:
** PositionalArgumentBuilder
*/

#include "PositionalArgumentBuilder.hpp"

namespace shell {
namespace command {
PositionalArgumentBuilder & PositionalArgumentBuilder::name(
    const std::string &name) noexcept
{
    _argument.name = name;

    return *this;
}

PositionalArgumentBuilder & PositionalArgumentBuilder::description(
    const std::string &description) noexcept
{
    _argument.description = description;

    return *this;
}

PositionalArgumentBuilder & PositionalArgumentBuilder::defaultValue(
    const std::string &defaultValue) noexcept
{
    _argument.defaultValue = defaultValue;

    return *this;
}

PositionalArgumentBuilder & PositionalArgumentBuilder::required() noexcept
{
    _argument.required = true;

    return *this;
}

PositionalArgument PositionalArgumentBuilder::build() const noexcept
{
    return _argument;
}
} // namespace command
} // namespace shell
